### PR TITLE
colmena: init at 0.2.0

### PR DESCRIPTION
**Pull request by zhaofengli**
###### Motivation for this change

[Colmena](https://zhaofengli.github.io/colmena/0.2) is a simple, stateless NixOS deployment tool.

This is the first stable release of the tool, submitted here so it can be easily installed from Nixpkgs.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/admin/colmena/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{ stdenv, lib, rustPlatform, fetchFromGitHub }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "colmena";
+  version = "0.2.0";
+
+  # We guarantee CLI and Nix API stability for the same minor version
+  apiVersion = builtins.concatStringsSep "." (lib.take 2 (lib.splitString "." version));
```
Comment (aanderse) [inline pkgs/tools/admin/colmena/default.nix:8]: Is this something we want to put in `passthru` instead?

File: pkgs/tools/admin/colmena/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  postInstall = lib.optionalString (stdenv.hostPlatform == stdenv.buildPlatform) ''
+    mkdir completions
+    for shell in bash fish zsh; do
+      $out/bin/colmena gen-completions $shell > completions/$shell
+    done
+
+    mkdir -p "$out/share/"{bash-completion/completions,fish/vendor_completions.d,zsh/site-functions}
+    cp completions/bash $out/share/bash-completion/completions/colmena
+    cp completions/fish $out/share/fish/vendor_completions.d/colmena.fish
+    cp completions/zsh $out/share/zsh/site-functions/_colmena
```
Comment (legendofmiracles) [inline pkgs/tools/admin/colmena/default.nix:28]:
```suggestion
installShellCompletion --cmd colmena \
	--bash <($out/bin/colmena gen-completions bash) \
	--zsh <($out/bin/colmena gen-completions zsh) \
	--fish <$out/bin/colmena gen-completions fish) 
```

File: pkgs/tools/admin/colmena/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
+{ stdenv, lib, rustPlatform, fetchFromGitHub }:
```
Comment (legendofmiracles) [inline pkgs/tools/admin/colmena/default.nix:1]:
```suggestion
{ stdenv, lib, rustPlatform, fetchFromGitHub, installShellFiles }:
```

File: pkgs/tools/admin/colmena/default.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+  src = fetchFromGitHub {
+    owner = "zhaofengli";
+    repo = "colmena";
+    rev = "v${version}";
+    sha256 = "sha256-WY8SYapnDcfaoLr1iFgwc9/E7xSfOFN2AvMDpk74AI8=";
+  };
+
+  cargoSha256 = "sha256-ZNSg3hXWKHNQ9yHJS1qW3tFYwzU4ZDa1N0yvoGLmWns=";
+
```
Comment (legendofmiracles) [inline pkgs/tools/admin/colmena/default.nix:18]:
```suggestion

nativeBuildInputs = [ installShellFiles ];

```

**Review by 0x4A6F — APPROVED** [review]
Built and tested on x86_64-linux and aarch64-linux.

**Comment by zhaofengli** [comment]
Added `testVersion` test in `passthru.tests.version`. Thanks @0x4A6F.

**Comment by aanderse** [comment]
🎉

**Pull request merged**